### ISSUE TITLE
eslint.nodePath not resulting in module resolving

### ISSUE THREAD

**Issue by felix-voicehub** [microsoft/vscode-eslint#729]
Hi,

My eslint is complaining about not finding the installed config to extend from:

> [Error - 12:19:26 PM] Error: Failed to load config "airbnb-base" to extend from.
Referenced from: /Users/ME/Desktop/dev/MYCODEFOLDER/client/.eslintrc.js
    at configMissingError (/usr/local/lib/node_modules/eslint/lib/cli-engine/config-array-factory.js:233:9)

This error shows up when opening the folder with my code in vscode, which has no node_modules, as everything is dockerized. There are `.eslintrc` files in that folder though.

I installed `node@10` using `brew`, I am on macOS.
`npm root -g` shows: `/usr/local/lib/node_modules`, so in my vscode settings, I put: 

> "eslint.nodePath": "/usr/local/lib/node_modules/"

The contents of that folder are:

```
eslint
eslint-config-airbnb-base
eslint-plugin-import
```

This is my install config:
```
npm config get prefix
/usr/local

which node
/usr/local/opt/node@10/bin/node

which npm
/usr/local/opt/node@10/bin/npm

which eslint
/usr/local/bin/eslint
```

What else could be wrong here for eslint to not work in vscode?

**Comment by dbaeumer** [comment]
@felix-voicehub does ESLint work for you in the terminal ?

The error indicates that `airbnb-base` plugin can't be loaded. Are you sure it is installed.

**Comment by felix-voicehub** [comment]
@dbaeumer `eslint` shows the commands, `eslint FILE` shows the same error:
> ESLint couldn't find the config "airbnb-base" to extend from. Please check that the name of the config is correct.

Extending from `airbnb-base` actually means this module: https://www.npmjs.com/package/eslint-config-airbnb-base, as can be seen in its readme. This module is globally installed as shown above.

**Comment by felix-voicehub** [comment]
I might have found the issue:

> Previously, ESLint loaded plugins relative to the location of the ESLint package itself. As a result, we suggested that users with global ESLint installations should also install plugins globally, and users with local ESLint installations should install plugins locally. However, due to a design bug, this strategy caused ESLint to randomly fail to load plugins and shareable configs under certain circumstances, particularly when using package management tools like lerna and Yarn Plug n’ Play.

> As a rule of thumb: **With ESLint v6, plugins should always be installed locally, even if ESLint was installed globally.** More precisely, ESLint v6 resolves plugins relative to the end user’s project by default, and always resolves shareable configs and parsers relative to the location of the config file that imports them.

> To address: If you use a global installation of ESLint (e.g. installed with npm install eslint --global) along with plugins, you should install those plugins locally in the projects where you run ESLint. If your config file extends shareable configs and/or parsers, you should ensure that those packages are installed as dependencies of the project containing the config file.

https://eslint.org/docs/user-guide/migrating-to-6.0.0#-plugins-and-shareable-configs-are-no-longer-affected-by-eslints-location

I was hoping to avoid this, but anyway, installing it in the working directory does solve this issue.

**Comment by dbaeumer** [comment]
@felix-voicehub great to hear. I will close the issue then. The ESLint v6 migration does cause issue mainly caused by ESLint itself (not the extension). Fort details you might want to look at #696
